core/simulations/helpers/promptToGetNumber.ts

```diff
@@ -0,0 +1,31 @@
+import prompts from 'prompts';
+
+import hre from 'hardhat';
+
+interface BlockNumberPromt {
+    title?: string;
+    initial?: number;
+    max?: number;
+    min?: number;
+}
+const promptToGetNumber = async (params?: BlockNumberPromt) => {
+    const title: string = params?.title || 'Block number to fork from';
+    const min = params?.min || 0;
+    const initial = params?.initial;
+    const max = params?.initial;
+    // only compute latest block if we need to fill values
+    const latestBlock = max && initial ? undefined : await hre.ethers.provider.getBlockNumber();
+    const { number } = await prompts([
+        {
+            type: 'number',
+            name: 'number',
+            message: title,
+            initial: initial || latestBlock,
```

Comment:
Nit: please prefer `??` over `||` when possible. Currently, if `initial` is set to 0 it wouldn't be processed correctly

Also
```suggestion
            initial: initial || max,
```